src/PointwiseFunctions/AnalyticData/ScalarTensor/KerrSphericalHarmonic.cpp

```diff
@@ -0,0 +1,134 @@
+
+#include <array>
+#include <cmath>
+#include <complex>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"     // IWYU pragma: keep
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/SwshInterpolation.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+// IWYU pragma:  no_include "DataStructures/Tensor/TypeAliases.hpp"
+
+namespace ScalarTensor::AnalyticData {
+
+KerrSphericalHarmonic::KerrSphericalHarmonic(
+    double mass, const std::array<double, 3>& dimensionless_spin,
+    double amplitude, double radius, double width,
+    std::pair<size_t, int> mode) {
+  mass_ = mass;
+  dimensionless_spin_ = dimensionless_spin;
+  amplitude_ = amplitude;
+  radius_ = radius;
+  width_sq_ = square(width);
+  mode_ = std::move(mode);
+  background_spacetime_ =
+      gr::Solutions::KerrSchild{mass_, dimensionless_spin_,
+                                // Center
+                                std::array<double, 3>{{0.0, 0.0, 0.0}}};
+}
+
+std::unique_ptr<evolution::initial_data::InitialData>
+KerrSphericalHarmonic::get_clone() const {
+  return std::make_unique<KerrSphericalHarmonic>(*this);
+}
+
+KerrSphericalHarmonic::KerrSphericalHarmonic(CkMigrateMessage* msg)
+    : InitialData(msg) {}
+
+void KerrSphericalHarmonic::pup(PUP::er& p) {
+  InitialData::pup(p);
+  p | mass_;
+  p | dimensionless_spin_;
+  p | amplitude_;
+  p | radius_;
+  p | width_sq_;
+  p | mode_;
+  p | background_spacetime_;
+}
+
+template <typename DataType>
+tuples::TaggedTuple<CurvedScalarWave::Tags::Psi>
+KerrSphericalHarmonic::variables(
+    const tnsr::I<DataType, 3>& x,
+    tmpl::list<CurvedScalarWave::Tags::Psi> /*meta*/) const {
+  return {make_with_value<Scalar<DataType>>(x, 0.0)};
+}
+
+template <typename DataType>
+tuples::TaggedTuple<CurvedScalarWave::Tags::Phi<3_st>>
+KerrSphericalHarmonic::variables(
+    const tnsr::I<DataType, 3>& x,
+    tmpl::list<CurvedScalarWave::Tags::Phi<3_st>> /*meta*/) const {
+  return {make_with_value<tnsr::i<DataType, 3>>(x, 0.0)};
+}
+
+template <typename DataType>
+tuples::TaggedTuple<CurvedScalarWave::Tags::Pi>
+KerrSphericalHarmonic::variables(
+    const tnsr::I<DataType, 3>& x,
+    tmpl::list<CurvedScalarWave::Tags::Pi> /*meta*/) const {
+  Scalar<DataType> pi =
+      make_with_value<Scalar<DataType>>(x, 0.0);
+  get(pi) += get(magnitude(x)) - radius_;
+  get(pi) = exp(-get(pi) * get(pi) / width_sq_);
+  const Spectral::Swsh::SpinWeightedSphericalHarmonic spherical_harmonic(
+      0, mode_.first, mode_.second);
+  const auto theta = atan2(hypot(x[0], x[1]), x[2]);
+  const auto phi = atan2(x[1], x[0]);
+  get(pi) *= real(spherical_harmonic.evaluate(theta, phi, sin(theta / 2.),
```

Comment:
`theta / 2.` -> `0.5 * theta` Division is ~ 25x slower than multiplication, and doing scalar math to the farthest left allows Blaze to do some optimizations (not always and I think not here, but just a good habit)